oracles/engine.go

```diff
@@ -64,6 +64,19 @@ func NewEngine(
 	return e
 }
 
+// ListensToPubKeys checks if the public keys from provided OracleData are among the keys
+// current OracleSpecs listen to.
+func (e *Engine) ListensToPubKeys(data OracleData) error {
+	_, err := e.subscriptions.filterSubscribers(func(spec OracleSpec) (bool, error) {
+		if !spec.matchPubKeys(data) {
+			return false, fmt.Errorf("provided data public keys do not match currently active in specs")
+		}
+		return true, nil
+	})
+
+	return err
+}
+
```

Comment:
Overall, this is very good. I think you can optimize this by implementing a `hasAnySubscribers() bool` method on the subscriptions structure, so it doesn't iterate over all the specs. We just needs to know if _at least one_ matches.